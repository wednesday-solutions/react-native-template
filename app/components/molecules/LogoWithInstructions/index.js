import React from 'react';
import { View, Image } from 'react-native';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { fonts, images } from '@themes';
import T from '@atoms/T';

const styles = {
  parentView: {
    marginTop: 30,
    marginBottom: 30
  },
  logoContainer: {
    width: '100%',
    height: 150,
    marginBottom: 25
  },
  logo: {
    width: '100%',
    height: '100%',
    alignSelf: 'center'
  }
};

const Instructions = styled(T)`
  ${fonts.style.standard()};
  text-align: center;
  margin-bottom: 5;
  font-style: italic;
`;

function LogoWithInstructions({ instructions }) {
  return (
    <View>
      <View style={styles.logoContainer}>
        <Image
          style={styles.logo}
          source={images.wednesdayLogo}
          resizeMode="contain"
        />
      </View>
      <Instructions text={instructions} />
    </View>
  );
}

LogoWithInstructions.propTypes = {
  instructions: PropTypes.string
};

export default LogoWithInstructions;
