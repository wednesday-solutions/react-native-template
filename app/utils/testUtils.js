import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import createStore from 'app/rootReducer';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@atoms/LanguageProvider';

export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});
export const renderWithIntl = (children, renderFunction = render) =>
  renderFunction(
    <IntlProvider
      locale={DEFAULT_LOCALE}
      messages={translationMessages[DEFAULT_LOCALE]}
    >
      {children}
    </IntlProvider>
  );

export const renderProvider = children => {
  const { store } = createStore();
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        {children}
      </ConnectedLanguageProvider>
    </Provider>
  );
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
