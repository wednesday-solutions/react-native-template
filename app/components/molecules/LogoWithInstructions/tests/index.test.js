/**
 *
 * Tests for LogoWithInstructions
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { renderWithI18next } from '@utils/testUtils';
import LogoWithInstructions from '../index';

describe('<LogoWithInstructions />', () => {
  it('Should render and match the snapshot', () => {
    const baseElement = renderWithI18next(<LogoWithInstructions />);
    expect(baseElement).toMatchSnapshot();
  });
  it('should render the instructions pased as props', () => {
    const instructions = 'PRESS CMD + D for iOS';
    const { getByText } = renderWithI18next(
      <LogoWithInstructions instructions={instructions} />
    );
    expect(getByText(instructions)).toBeTruthy();
  });
});
