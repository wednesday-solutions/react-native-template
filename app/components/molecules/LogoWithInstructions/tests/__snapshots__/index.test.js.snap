// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoWithInstructions /> Should render and match the snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "height": 150,
        "marginBottom": 25,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../app/assets/images/wednesday-logo.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "fontStyle": "italic",
          "fontWeight": "normal",
          "marginBottom": 5,
          "textAlign": "center",
        },
      ]
    }
  />
</View>
`;
